### DISCUSSION TITLE
frozen application in flet

### DISCUSSION THREAD

**Discussion by ingdesarr1** [flet-dev/flet#4060] [Q&A]
### Question

Good afternoon. I'm develop a app with flet, using a NavigationRail and multiple pages, to have many interfaces for the user. I'm using a lot of custom controls, and all the pages work fine, but recently, the app is freezing when switching between pages multiple times, but no obvious errors appear. Using the flet debug, the error message shown below appeared. What does this message mean and how can I resolve it? Thank you

![image](https://github.com/user-attachments/assets/da6baca5-8fb4-474a-99ee-0f22b3293fd8)

### Error message

```bash
DEBUG:flet_runtime:_on_message: {"action":"pageEventFromWeb","payload":{"eventTarget":"page","eventName":"error","eventData":"RangeError: Invalid value: Not in inclusive range 0..23: 25"}}
DEBUG:flet_runtime:sent to TCP: 1018
DEBUG:flet_runtime:_on_message: {"action":"pageEventFromWeb","payload":{"eventTarget":"page","eventName":"error","eventData":"RangeError: Invalid value: Not in inclusive range 0..23: 25"}}
DEBUG:flet_runtime:_on_message: {"action":"pageEventFromWeb","payload":{"eventTarget":"page","eventName":"error","eventData":"RangeError: Invalid value: Not in inclusive range 0..23: 25"}}
DEBUG:flet_runtime:_on_message: {"action":"pageEventFromWeb","payload":{"eventTarget":"page","eventName":"app_lifecycle_state_change","eventData":"inactive"}}
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by tanmay-bhatgare** [comment]
Have you tried updating flet? Would you mind sharing your code for better understanding?

**Comment by ap4499** [comment]
What Flet version are you on?

Could you share some code? It looks like something might be trying to navigate to a value that doesn't exist (likely a page in whatever array you have setup).

It is unusual that navigation seems to work, and then suddenly stop. Is there a particular page that you can't navigate to? Or a particular sequence that is breaking it? 

Normally I have my page change logic wrapped up in a class and I haven't noticed this problem.. but I would suspect that based solely on that error message, and without knowledge of your code, that you may be trying to hardcode navigate to page 25 and that that page doesn't exist?

**Reply by tanmay-bhatgare** [reply]
@ap4499 see mate this is off topic but can you tell me what's the best way to navigate in flet, I am trying very hard to understand how to do that means I can't navigate back using `on_view_pop` function provided by flet. Did you use same approach or different one? Thanks!

**Reply by ap4499** [reply]
This is the example that I used as the basis for navigation in my app. You should be able to add a back button and attach it to a on_view_pop method.

https://github.com/flet-dev/examples/blob/main/python/community/simple_desktop_layout/simple_desktop_layout.py

I don't actually use a back button type functionality as I have a traditional desktop app and most don't have that kind of UI experience.

**Comment by ndonkoHenri** [comment]
Please read this: https://github.com/flet-dev/flet/issues/1922#issuecomment-2138648179

**Reply by ingdesarr1** [reply]
In my program, I'm using this class to change the pages usign the NavigationRail and the on_route_change in the main. Is the correct way to do that, or maybe need other way? All the pages return a Container
```
class Router:

    def __init__(self,page):
        super().__init__()
        self.__lock = Lock()
        self.page = page
        self.ft = ft
        self.barra = RailBar(self.page)
        self.routes = {
            '/login': Login(self.page)
            # '/login': view()
        }

        self.space = ft.Container(
            content=self.routes['/login'],
            expand=True,
            width=float("inf"),
            height=float("inf")
            )
        
        self.body = ft.Row(
            controls=[
                self.barra,
                self.space,
            ],
            expand=True,
            width=float("inf"),
            height=float("inf")
            )
        
    def route_change(self, route):
        with self.__lock:
            if self.page.route == '/':
                self.routes = {
                    '/begin': Begin(self.page),
                    '/': Index(self.page),
                    '/iprocess': Process(self.page),
                    '/users': Users(self.page),
                    '/login': Login(self.page),
                    # '/logout': Logout(page),
                    '/externos': Clients(self.page),
                    '/parameters': Parameters(self.page),
                    '/reports': Reports(self.page),
                    # '/despachos': Despachos(self.page),
                    '/inventario': Inventario(self.page),
                }
            if self.page.route == '/logout':
                self.page.dialog = Logout(self.page)
                # self.barra.selected_index = 0

            elif self.page.route == '/login':
                self.routes = {
                    '/login': Login(self.page)
                }
                self.barra.content.selected_index = 0
                self.space.content = self.routes[route.route]
                self.barra.visible = False
                self.page.appbar.visible = False
            else:
                self.barra.content.leading = opciones.opciones()
                self.space.content = self.routes[route.route]
                self.barra.visible = True
                self.page.appbar.visible = True
            self.page.update()
```

**Reply by ndonkoHenri** [reply]
Will be nice if you could share some runnable code, which reproduces the issue.

**Reply by ingdesarr1** [reply]
@ndonkoHenri  I discovered that the root of this problem, "RangeError: Invalid value: Not in inclusive range 0..23: 25", maybe caused by the objects append in the overlay. When I changed the page, in some of this pages I have elements like datepickers, and changing fast between pages, the app freeze. if I clean the overlay before change page, the issue is not generated. I use self.page.overlay.clear() to fix this. Is a correct way to fix this? Thanks
